src/main/java/sx/blah/discord/api/internal/DiscordClientImpl.java

```diff
@@ -211,30 +212,64 @@ public IUser getOurUser() {
 		return ourUser;
 	}
 
-	@Override
-	public List<IRegion> getRegions() {
-		if (REGIONS.isEmpty()) {
-			VoiceRegionObject[] regions = REQUESTS.GET.makeRequest(
-					DiscordEndpoints.VOICE+"regions", VoiceRegionObject[].class);
+	private synchronized void loadStandardRegions() {
+		if (regions.isEmpty()) { // Guarantee so standard regions are first
+			VoiceRegionObject[] regionObjects = RequestBuffer.request(() ->
+					(VoiceRegionObject[]) REQUESTS.GET.makeRequest(
+							DiscordEndpoints.VOICE + "regions", VoiceRegionObject[].class)).get();
+
+			Arrays.stream(regionObjects)
+					.map(DiscordUtils::getRegionFromJSON)
+					.forEach(r -> regions.putIfAbsent(r.getID(), r));
+		}
+	}
 
-			Arrays.stream(regions)
+	public synchronized IRegion getGuildRegion(Guild guild) {
+		loadStandardRegions();
+		IRegion region = regions.get(guild.getRegionID());
+
+		if (region == null) { // New region types means Discord has updated
+			VoiceRegionObject[] regionObjects = RequestBuffer.request(() ->
+					(VoiceRegionObject[]) REQUESTS.GET.makeRequest(
+							DiscordEndpoints.GUILDS + guild.getStringID() + "/regions", VoiceRegionObject[].class)).get();
+
+			Arrays.stream(regionObjects)
 					.map(DiscordUtils::getRegionFromJSON)
-					.forEach(REGIONS::add);
+					.forEach(r -> regions.putIfAbsent(r.getID(), r));
+
+			region = regions.get(guild.getRegionID());
 		}
 
-		return REGIONS;
+		return region;
+	}
+
+	private void loadAllRegions() {
+		loadStandardRegions();
+
+		shards.stream()
+				.map(IShard::getGuilds)
+				.flatMap(List::stream)
+				.map(g -> (Guild) g)
+				.forEach(this::getGuildRegion);
+	}
+
+	@Override
+	public List<IRegion> getRegions() {
+		loadAllRegions();
+		return new ArrayList<>(regions.values());
 	}
 
 	@Override
 	public IRegion getRegionByID(String regionID) {
-		try {
-			return getRegions().stream()
-					.filter(r -> r.getID().equals(regionID))
-					.findAny().orElse(null);
-		} catch (RateLimitException | DiscordException e) {
-			Discord4J.LOGGER.error(LogMarkers.API, "Discord4J Internal Exception", e);
+		loadStandardRegions();
```

Comment:
This is not needed

Reply:
Should not need to call more endpoints than needed. If the regionID is part of the standard region, then it's best to make 1 call, versus multiple calls it would have to do for `loadAllRegions`

Reply:
This is not needed because `loadAllRegions` calls `loadStandardRegions`

Reply:
It is needed because of the following situation:
You call this method before the first invocation of `loadStandardRegions` happens and you're passing in an ID that's a standard region.
If this method is removed, what will happen is it will call `loadAllRegions` because it's obviously null. Which means this will call not only just `loadStandardRegions` but all call the guild endpoint if a VIP guilds existed.
If this method stays in, then it will only do the standard call because the region wouldn't have returned null.

Reply:
So call `getGuildRegion` rather than `getAllRegions`